hooks/commit-msg.sample

```diff
@@ -14,3 +14,25 @@ if [ -z "$EMAIL" ]; then
 fi
 
 git interpret-trailers --if-exists doNothing --trailer "Signed-off-by: $NAME <$EMAIL>" --in-place "$1"
+
+# Commits message validation
+Color_Off='\033[0m'
+BRed="\033[1;31m"         # Red
+BGreen="\033[1;32m"       # Green
+BYellow="\033[1;33m"      # Yellow
+BBlue="\033[1;34m"        # Blue
+
+MSG_FILE=$1
+FILE_CONTENT="$(cat $MSG_FILE)"
+export REGEX="\[feature\] \#[[:digit:]]+\: .+$|^\[refactor\](( \#)([[:digit:]]+))?\: .+$|^\[fix\](( \#)([[:digit:]]+))?\: .+$|^\[documentation\](( \#)([[:digit:]]+))?\: .+$|^\[ci\](( \#)([[:digit:]]+))?\: .+$|^[M-Mm-m]erge*.+$"
+export ERROR_MSG="Commit message format must match the regex \"${REGEX}\""
+if [[ $FILE_CONTENT =~ $REGEX ]]; then
+ printf "${BGreen}Correct commit message!${Color_Off}"
+else
+  printf "${BRed}Wrong commit message!${BBlue}\"$FILE_CONTENT\"\n"
+ printf "${BYellow}$ERROR_MSG\n"
+ printf "commit-msg hook failed (add --no-verify to bypass)\n"
```

Comment:
Excellent use of confusing indentation. Normally the standard is either four or two spaces.